geonode/base/bbox_utils.py

```diff
@@ -78,27 +89,149 @@ def filter_bbox(queryset, bbox):
             _bbox_index += 1
         bboxes[_bbox_index].append(_y)
 
+    search_queryset = None
     for _bbox in bboxes:
         _bbox = list(map(Decimal, _bbox))
 
         # Return all layers when the search extent exceeds 360deg
         if abs(_bbox[0] - _bbox[2]) >= 360:
             return queryset.all()
```

Comment:
@afabiani I know this code was already there, but given that we're improving it :
1. how to valid WGS84 coords could span more then 360 degrees?
2.  I don't understand why we only consider lat / x coordinates to decide that we should return everything. It might be only a strip all around the world but just a few degrees "high", don't you think?

Reply:
@afabiani let's remove that condition.

Reply:
Done